amap: add page (#9201)
tldr-pages/tldr: pages/linux/amap.md

# amap

> Next-generation scanning tool for pentesters.

> > More information: <https://www.thc.org/>.

Reviewer: ```suggestion
> More information: <https://www.thc.org>.
```

> - Scan port 80 on a target host and print ASCII banner response, omit closed ports:

Reviewer: ```suggestion
- Scan the port 80 on a target host and print ASCII banner response, omitting closed ports:
```

> - Scan port 80 on target IP address and print ASCII banner response, omit closed ports, verbosely:

Reviewer: But this command doesn't have a `-v` argument
```suggestion
- Scan port 80 on a target host and print ASCII banner response, omit closed ports, with detailed information:
```

Reviewer: Which one fits better?
```suggestion
- Scan port 80 on a target host and print ASCII banner response, omit closed ports:
```

> - Scan port 80 on a target host and print ASCII banner response, omit closed ports:
> 
> `amap -bqv {{ip_or_hostname}} {{80}}`

Reviewer 2: ```suggestion
- Scan port 80 on a target host and print ASCII banner response, omit closed ports:

`amap -bq --verbose {{ip|hostname}} {{80}}`
```

Warning: [mnemonics can't be suggested as parsing `amap` man page failed: no `-b`, `-q` options found, possible broken man page](https://manned.org/amap.1).
Mandatory fix: [long option must be used if it's portable](https://github.com/tldr-pages/tldr/blob/main/contributing-guides/style-guide.md#option-syntax).
Mandatory fix: [`or` keyword is undefined for anything expect `file_or_directory`, wrong syntax used](https://github.com/tldr-pages/tldr/blob/main/contributing-guides/style-guide.md#paths).

> - Scan port 1-1024 and capture banners:

Reviewer: ```suggestion
- Scan the ports 1-1024 and capture banners:
```

> - Scan port 1-1024 and capture banners:
> 
> `amap -bqv {{ip_or_hostname}} {{1-1024}} -B`

Reviewer 2: ```suggestion
- Scan a specific port and capture banners:

`amap -bqv {{ip|hostname}} {{1..1024}} -B`
```

Almost the same issue as above.
Warning: mnemonics can't be suggested for options as parsing man page failed and possibly several short options differing just with case detected.
Optional fix: `specific` may sound better.
Optional fix: `..` can be used for ranges, our Emacs extension accepts just such syntax as correct (standardization).

> `amap -bvq {{ip_or_hostname}} {{1-1024}} -B`

Reviewer: We prefer long arguments if available instead of `-B`

Author: Unfortunately application does not support long arguments

> - Scan port 1-1024 and analyze the banners results efficiently:

Reviewer: How is an `analyze` non-`efficiently`? :thinking:

> - Scan port 1-1024 and analyze the banners results efficiently:
> 
> `amap -bqv {{ip_or_hostname}} {{1-1024}} -A`

Reviewer 2: ```suggestion
- Scan a specific port and analyze the banners results efficiently:

`amap -bqv {{ip|hostname}} {{1..1024}} -A`
```
The same issue as above.

> - Scan port 1-1024 without banner grabbing and return open ports:

Reviewer: ```suggestion
- Scan the ports 1-1024 without banner grabbing and display open ports:
```

> - Scan port 1-1024 without banner grabbing and return open ports:
> 
> `amap {{ip_or_hostname}} {{1-1024}} -P`

Reviewer 2: ```suggestion
- Scan a specific port without banner grabbing and return open ports:

`amap {{ip|hostname}} {{1..1024}} -P`
```
The same issue as above.
